src/__tests__/browserSuites/push-flag-sets.spec.js

```diff
@@ -0,0 +1,262 @@
+import { SplitFactory } from '../..';
+import EventSourceMock, { setMockListener } from '../testUtils/eventSourceMock';
+window.EventSource = EventSourceMock;
+
+import notification1 from '../mocks/message.SPLIT_UPDATE.FS.1.json';
+import notification2 from '../mocks/message.SPLIT_UPDATE.FS.2.json';
+import notification3 from '../mocks/message.SPLIT_UPDATE.FS.3.json';
+import notification4None from '../mocks/message.SPLIT_UPDATE.FS.4None.json';
+import notification4 from '../mocks/message.SPLIT_UPDATE.FS.4.json';
+import notification5 from '../mocks/message.SPLIT_UPDATE.FS.5.json';
+import notificationKill from '../mocks/message.SPLIT_UPDATE.FS.kill.json';
+
+import authPushEnabled from '../mocks/auth.pushEnabled.nicolas@split.io.json';
+
+const baseUrls = {
+  sdk: 'https://sdk.baseurl',
+  auth: 'https://auth.baseurl/api'
+};
+
+const baseConfig = {
+  core: {
+    authorizationKey: '<fake-token>',
+    key: 'nicolas@split.io'
+  },
+  urls: baseUrls,
+};
+
+const MILLIS_FIRST_SPLIT_UPDATE_EVENT = 100;
+const MILLIS_SECOND_SPLIT_UPDATE_EVENT = 200;
+const MILLIS_THIRD_SPLIT_UPDATE_EVENT = 300;
+const MILLIS_FOURTH_SPLIT_UPDATE_EVENT = 400;
+
+export function testFlagSets(fetchMock, t) {
+  fetchMock.reset();
+
+  fetchMock.get(baseUrls.sdk + '/mySegments/nicolas%40split.io', { status: 200, body: { 'mySegments': [] } });
+
+  fetchMock.get(baseUrls.auth + '/v2/auth?users=nicolas%40split.io', function () {
+    return { status: 200, body: authPushEnabled };
+  });
+  fetchMock.get(baseUrls.sdk + '/splitChanges?since=-1',  function () {
+    return { status: 200, body: { splits: [], since: -1, till: 0}};
+  });
+  fetchMock.get(baseUrls.sdk + '/splitChanges?since=0',  function () {
+    return { status: 200, body: { splits: [], since: 0, till: 1 }};
+  });
+  fetchMock.get(baseUrls.sdk + '/splitChanges?since=-1&sets=set_1,set_2',  function () {
+    return { status: 200, body: { splits: [], since: -1, till: 0 }};
+  });
+  fetchMock.get(baseUrls.sdk + '/splitChanges?since=0&sets=set_1,set_2',  function () {
+    return { status: 200, body: { splits: [], since: 0, till: 1 }};
+  });
+
+
+  const configWithSets = {
+    ...baseConfig,
+    sync: {
+      splitFilters: [{type: 'bySet', values: ['set_1', 'set_2']}]
+    }
+  };
+
+  t.test(async (assert) => {
+
+    assert.plan(3);
+    let splitio, client, manager = [];
+
+    setMockListener((eventSourceInstance) => {
+      eventSourceInstance.emitOpen();
+
+      setTimeout(() => {
+        assert.deepEqual(manager.splits(), [], '1 - initialized without flags');
+        client.once(client.Event.SDK_UPDATE, async () => {
+          assert.equal(manager.splits().length, 1, '1 - update is processed and the flag is stored');
+          await client.destroy();
+          assert.equal(eventSourceInstance.readyState, EventSourceMock.CLOSED, '1 - streaming is closed after destroy');
```

Comment:
remove line 75, because this is something already validated in other test.
Remove it also on other tests too.